packages/editor/src/components/editor-help/index.native.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import { View, Text } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { BottomSheet, PanelBody } from '@wordpress/components';
+import { useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { withPreferredColorScheme } from '@wordpress/compose';
+import {
+	helpFilled,
+	plusCircleFilled,
+	alignJustifyAlt,
+	trashFilled,
+	cogAlt,
+} from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+import HelpDetailNavigationScreen from './help-detail-navigation-screen.js';
+import HelpTopicRow from './help-topic-row.js';
+
+const HELP_TOPICS = [
+	{ label: __( 'What is a block?' ), icon: helpFilled },
+	{ label: __( 'Add blocks' ), icon: plusCircleFilled },
+	{ label: __( 'Move blocks' ), icon: alignJustifyAlt },
+	{ label: __( 'Remove blocks' ), icon: trashFilled },
+	{ label: __( 'Customize blocks' ), icon: cogAlt },
+];
+
+function EditorHelpTopics( { isVisible, onClose, getStylesFromColorScheme } ) {
+	const bottomSheetHeaderTitleStyle = getStylesFromColorScheme(
+		styles.bottomSheetHeaderTitle,
+		styles.bottomSheetHeaderTitleDark
+	);
+
+	return useMemo( () => (
+		<BottomSheet
+			isVisible={ isVisible }
+			onClose={ onClose }
+			hideHeader
+			hasNavigation
+		>
+			<BottomSheet.NavigationContainer animate main>
+				<BottomSheet.NavigationScreen name={ 'Topics' }>
+					<View style={ styles.bottomSheetHeader }>
+						<Text
+							style={ bottomSheetHeaderTitleStyle }
+							maxFontSizeMultiplier={ 3 }
+						>
+							{ __( 'How to edit your site' ) }
+						</Text>
+					</View>
+					<View style={ styles.separator } />
+					<PanelBody
+						title={ __( 'The basics' ) }
+						style={ styles.sectionContainer }
+					>
+						{ /* Print out help topics */ }
+						{ HELP_TOPICS.map( ( topic ) => {
+							const key = topic.label;
+							return (
+								<HelpTopicRow
+									key={ key }
+									label={ topic.label }
+									icon={ topic.icon }
+								/>
+							);
+						} ) }
+					</PanelBody>
+				</BottomSheet.NavigationScreen>
+				{ /* Print out help detail screens */ }
+				{ HELP_TOPICS.map( ( topic ) => {
+					const key = topic.label;
+					return (
+						<HelpDetailNavigationScreen
+							key={ key }
+							name={ topic.label }
+						/>
+					);
+				} ) }
```

Comment:
We briefly discussed it in chat, but I put together an example of using `BottomSheet.SubSheet`. You can [view the diff here](https://github.com/WordPress/gutenberg/compare/rnmobile/editor-onboarding-basic-help-2...rnmobile/editor-onboarding-basic-help-2-with-subsheet?expand=1), if you'd like to consider leveraging it. I did not heavily test it. 

I also noticed that existing usage of `BottomSheet.SubSheet` outputs an error. It doesn't appear to cause issue in the UI presented, but wanted to note it in case we choose to use `BottomSheet.SubSheet`. I believe [this effect](https://github.com/WordPress/gutenberg/blob/0d34c1c46a49e585e2b7bea1df9821802c4708a8/packages/components/src/mobile/bottom-sheet/sub-sheet/index.native.js#L27) needs to be wrapped in `useEffect`, [similar to `BlockMobileToolbar`](https://github.com/WordPress/gutenberg/blob/0d34c1c46a49e585e2b7bea1df9821802c4708a8/packages/block-editor/src/components/block-mobile-toolbar/index.native.js#L56-L62), which was fixed in https://github.com/WordPress/gutenberg/commit/85dc86c3e36f39dab901466388b24d76ec236c71.

<details><summary>SubSheet Error</summary>

```
 ERROR  Warning: Cannot update during an existing state transition (such as within `render`). Render methods should be a pure function of props and state.
    in RCTSafeAreaView (at SafeAreaView.js:51)
    in SafeAreaView (at sub-sheet/index.native.js:24)
    in SlotComponent (at slot.js:91)
    in Slot (at slot-fill/index.native.js:16)
    in Slot (at slot-fill/index.native.js:23)
    in BottomSheetSubSheetSlot (at container.native.js:55)
    in RCTView (at View.js:34)
    in View (at navigation-screen.native.js:86)
    in RCTView (at View.js:34)
    in View (at TouchableHighlight.js:289)
    in TouchableHighlight (at TouchableHighlight.js:352)
    in Unknown (at navigation-screen.native.js:85)
    in RCTScrollContentView (at ScrollView.js:1107)
    in RCTScrollView (at ScrollView.js:1238)
    in ScrollView (at ScrollView.js:1264)
    in ScrollView (at navigation-screen.native.js:84)
    in BottomSheetNavigationScreen (at container.native.js:52)
    in StaticContainer
    in EnsureSingleNavigator (at SceneView.tsx:104)
    in SceneView (at useDescriptors.tsx:150)
    in RCTView (at View.js:34)
    in View (at CardContainer.tsx:198)
    in RCTView (at View.js:34)
    in View (at CardContainer.tsx:197)
    in RCTView (at View.js:34)
    in View (at CardSheet.tsx:33)
    in CardSheet (at Card.tsx:543)
    in RCTView (at View.js:34)
    in View (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at Card.tsx:525)
    in PanGestureHandler (at GestureHandler.native.tsx:13)
    in PanGestureHandler (at Card.tsx:519)
    in RCTView (at View.js:34)
    in View (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at Card.tsx:515)
    in RCTView (at View.js:34)
    in View (at Card.tsx:509)
    in Card (at CardContainer.tsx:167)
    in CardContainer (at CardStack.tsx:552)
    in RCTView (at View.js:34)
    in View (at Screens.tsx:70)
    in MaybeScreen (at CardStack.tsx:545)
    in RCTView (at View.js:34)
    in View (at Screens.tsx:48)
    in MaybeScreenContainer (at CardStack.tsx:455)
    in CardStack (at StackView.tsx:437)
    in KeyboardManager (at StackView.tsx:435)
    in RNCSafeAreaProvider (at SafeAreaContext.tsx:76)
    in SafeAreaProvider (at SafeAreaProviderCompat.tsx:42)
    in SafeAreaProviderCompat (at StackView.tsx:432)
    in RCTView (at View.js:34)
    in View (at StackView.tsx:431)
    in StackView (at createStackNavigator.tsx:82)
    in StackNavigator (at navigation-container.native.js:145)
    in EnsureSingleNavigator (at BaseNavigationContainer.tsx:376)
    in BaseNavigationContainer (at NavigationContainer.tsx:91)
    in ThemeProvider (at NavigationContainer.tsx:90)
    in NavigationContainer (at navigation-container.native.js:144)
    in RCTView (at View.js:34)
    in View (at navigation-container.native.js:132)
    in BottomSheetNavigationContainer (at container.native.js:46)
    in RCTView (at View.js:34)
    in View (at bottom-sheet/index.native.js:541)
    in RCTView (at View.js:34)
    in View (at keyboard-avoiding-view.native.js:118)
    in KeyboardAvoidingView (at bottom-sheet/index.native.js:518)
    in RCTView (at View.js:34)
    in View (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at createAnimatableComponent.js:599)
    in withAnimatable(View) (created by ReactNativeModal)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:107)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:134)
    in AppContainer (at Modal.js:228)
    in RCTView (at View.js:34)
    in View (at Modal.js:249)
    in VirtualizedListContextResetter (at Modal.js:247)
    in RCTModalHostView (at Modal.js:234)
    in Modal (created by ReactNativeModal)
    in ReactNativeModal (at bottom-sheet/index.native.js:489)
    in BottomSheet (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(BottomSheet) (at container.native.js:38)
    in BottomSheetSettings (at with-dispatch/index.js:96)
    in WithDispatch(BottomSheetSettings) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:43)
    in WithSelect(WithDispatch(BottomSheetSettings)) (at layout/index.native.js:172)
    in RCTView (at View.js:34)
    in View (at keyboard-avoiding-view/index.ios.js:123)
    in RCTView (at View.js:34)
    in View (at KeyboardAvoidingView.js:209)
    in KeyboardAvoidingView (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at keyboard-avoiding-view/index.ios.js:104)
    in KeyboardAvoidingView (at layout/index.native.js:160)
    in RCTSafeAreaView (at SafeAreaView.js:51)
    in SafeAreaView (at layout/index.native.js:137)
    in RCTView (at View.js:34)
    in View (at tooltip/index.native.js:268)
    in TooltipSlot (at layout/index.native.js:136)
    in Layout (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(Layout) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:43)
    in WithSelect(WithPreferredColorScheme(Layout)) (at editor.native.js:167)
    in BlockRefsProvider (at provider/index.native.js:28)
    in BlockEditorProvider (at with-registry-provider.js:24)
    in Unknown (at with-registry/index.js:23)
    in WithRegistryProvider(BlockEditorProvider) (at provider/index.js:108)
    in BlockContextProvider (at provider/index.js:107)
    in EntityProvider (at provider/index.js:106)
    in EntityProvider (at provider/index.js:105)
    in EditorProvider (at with-registry-provider.js:27)
    in Unknown (at with-registry/index.js:23)
    in WithRegistryProvider(EditorProvider) (at provider/index.native.js:305)
    in NativeEditorProvider (at with-dispatch/index.js:96)
    in WithDispatch(NativeEditorProvider) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:43)
    in WithSelect(WithDispatch(NativeEditorProvider)) (at editor.native.js:160)
    in SlotFillProvider (at editor.native.js:159)
    in Editor (at with-dispatch/index.js:96)
    in WithDispatch(Editor) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:43)
    in WithSelect(WithDispatch(Editor)) (at src/index.native.js:31)
    in Unknown (at renderApplication.js:47)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:107)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:134)
    in AppContainer (at renderApplication.js:40)
```

</details>

Reply:
Wow, thank you for all this extra work and research you did! I really appreciate it @dcalhoun . I had tried using the SubSheet before but couldn't get it to work for me. I implemented your version of it in 4b1e3b5b8d04dafb5e7a3a718c41f49aab3ffdff. I also fixed the `BottomSheet.SubSheet` render error in 4b1e3b5b8d04dafb5e7a3a718c41f49aab3ffdff. Please let me know what you think 👍

Reply:
You're welcome! Happy to do it. I think your new implementation looks great and helps us avoid redundant code for a common pattern of sub-sheets. Thanks!